web/js/components/timeline/timeline-controls/animation-button.js

```diff
@@ -1,38 +1,75 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { UncontrolledTooltip } from 'reactstrap';
 
-class AnimationButton extends PureComponent {
-  render() {
-    const className = 'button-action-group animate-button';
-    const { disabled, label, clickAnimationButton } = this.props;
-    const buttonId = 'animate-button';
-    const labelText = label || 'Set up animation';
-    return (
-      <div
-        onClick={clickAnimationButton}
-        className={disabled ? `wv-disabled-button ${className}` : className}
-        aria-label={labelText}
-      >
-        <div id={buttonId}>
-          <UncontrolledTooltip
-            placement="top"
-            target={buttonId}
-          >
-            {labelText}
-          </UncontrolledTooltip>
-          <FontAwesomeIcon icon="video" className="wv-animate" size="2x" />
-        </div>
+const AnimationButton = (props) => {
+  const {
+    disabled,
+    label,
+    clickAnimationButton,
+    breakpoints,
+    screenWidth,
+    isLandscape,
+    isPortrait,
+    isMobilePhone,
+    isMobileTablet,
+    isMobile,
+    hasSubdailyLayers,
+  } = props;
+
+  const subdailyID = hasSubdailyLayers ? '-subdaily' : '';
+  const buttonId = 'animate-button';
+  const labelText = label || 'Set up animation';
+
+  const getButtonClassName = () => {
+    if ((isMobilePhone && isPortrait) || (!isMobileTablet && screenWidth < 670 && hasSubdailyLayers) || (!isMobileTablet && screenWidth < 575 && !hasSubdailyLayers)) {
+      return `phone-portrait${subdailyID}`;
+    } if (isMobilePhone && isLandscape) {
+      return `phone-landscape${subdailyID}`;
+    } if ((isMobileTablet && isPortrait) || (!isMobilePhone && screenWidth < breakpoints.small)) {
+      return `tablet-portrait${subdailyID}`;
+    } if (isMobileTablet && isLandscape) {
+      return `tablet-landscape${subdailyID}`;
+    }
+  };
+
+  const className = getButtonClassName();
```

Comment:
I would prefer the "className" variable have a different name since "className" is a JSX term. btnClassName? btnClassSuffix?